service/src/main/kotlin/fi/espoo/evaka/occupancy/Occupancy.kt

```diff
@@ -384,19 +384,30 @@ private fun <K : OccupancyGroupingKey> Database.Read.calculateDailyOccupancies(
 
     val serviceNeeds = this.createQuery(
         """
-SELECT sn.placement_id, sno.occupancy_coefficient, daterange(sn.start_date, sn.end_date, '[]') AS period
+SELECT
+    sn.placement_id,
+    CASE
+        WHEN u.type && array['FAMILY', 'GROUP_FAMILY']::care_types[] THEN 1.75
+        WHEN extract(YEARS FROM age(ch.date_of_birth)) < 3 THEN coalesce(sno.occupancy_coefficient_under_3y, default_sno.occupancy_coefficient_under_3y)
+        ELSE sno.occupancy_coefficient
+    END AS occupancy_coefficient,
+    daterange(sn.start_date, sn.end_date, '[]') AS period
 FROM service_need sn
+JOIN placement pl ON sn.placement_id = pl.id
 JOIN service_need_option sno ON sn.option_id = sno.id
-WHERE sn.placement_id = ANY(:placementIds)
+JOIN service_need_option default_sno on pl.type = default_sno.valid_placement_type AND default_sno.default_option
+JOIN person ch ON ch.id = pl.child_id
+JOIN daycare u ON pl.unit_id = u.id
+WHERE sn.placement_id = ANY(:placementIds) AND ch.date_of_birth IS NOT NULL
 """
     )
         .bind("placementIds", placements.map { it.placementId }.toList().toTypedArray())
         .mapTo<ServiceNeed>()
         .groupBy { it.placementId }
 
     val defaultServiceNeedCoefficients =
-        this.createQuery("SELECT occupancy_coefficient, valid_placement_type FROM service_need_option WHERE default_option")
-            .map { row -> row.mapColumn<PlacementType>("valid_placement_type") to row.mapColumn<BigDecimal>("occupancy_coefficient") }
+        this.createQuery("SELECT occupancy_coefficient, occupancy_coefficient_under_3y, valid_placement_type FROM service_need_option WHERE default_option")
+            .map { row -> row.mapColumn<PlacementType>("valid_placement_type") to Pair(row.mapColumn<BigDecimal>("occupancy_coefficient"), row.mapColumn<BigDecimal>("occupancy_coefficient_under_3y")) }
```

Comment:
Naming the fields could be more readable when the values are used a few lines down from here:

```suggestion
            .map { row -> row.mapColumn<PlacementType>("valid_placement_type") to object {
                val occupancyCoefficient = row.mapColumn<BigDecimal>("occupancy_coefficient")
                val occupancyCoefficientUnder3y = row.mapColumn<BigDecimal>("occupancy_coefficient_under_3y")
              }
            }
```